src/lib/user/user-actor.ts

```diff
@@ -44,4 +47,42 @@ export class UserActor {
     const { userMapper } = this.config;
     return userMapper.mapUserRecordToUserPii(record);
   }
+
+  public async getDogList(): Promise<DogList> {
+    const { dbPool, dogMapper } = this.config;
+    const sql = `
+    SELECT
+      dog_id as "dogId",
+      dog_encrypted_oii as "dogEncryptedOii",
+      CASE
+        WHEN dog_breed = '' AND dog_weight_kg IS NULL THEN $2
+        WHEN dog_weight_kg < 20 THEN $4
+        WHEN dog_ever_pregnant = 'YES' THEN $5
+        WHEN dog_ever_received_transfusion = 'YES' THEN $5
```

Comment:
The PERMANENTLY_INELIGIBLE rules should be evaluated first.

Reply:
Fixed in https://github.com/barkbank/barkbank-pawtal/pull/70/commits/cb4a59165ae00d7b7db039aa6fbe40280320ec75